amp.php

```diff
@@ -71,6 +71,7 @@ function amp_after_setup_theme() {
 	}
 
 	add_action( 'init', 'amp_init' );
+	add_action( 'init', 'amp_add_widget_support' );
```

Comment:
Should this only be done if `amp_is_canonical()`?

Reply:
Hi @westonruter, good catch. It should at least check that it [is_amp_endpoint()](https://github.com/Automattic/amp-wp/blob/develop/includes/amp-helper-functions.php#L120). We wouldn't want these subclass widgets on a standard WP page.

Reply:
The theme should be styling whatever markup the the widgets in core (or the plugin) output. The AMP widget functionality you're adding here should only apply if `is_amp_endpoint()` because this will true both when viewing AMP in paired mode and when canonical.

Reply:
Thanks, @westonruter. You probably saw it, but [this commit](https://github.com/Automattic/amp-wp/pull/870/commits/74d727b1f96035d01a1db594502891cc4bc6ed7a) applies your suggestion. 

I had tried to wrap the action hook in the conditional:

```add_action( 'init', 'amp_add_widget_support' )```
But this produced a notice:
```
is_amp_endpoint was called <strong>incorrectly</strong>. is_amp_endpoint() was called before the &#39;parse_query&#39; hook was called.
```

Reply:
I've pushed a suggested change in dc3f5b4 to fix.